cluster/src/test/scala/org/apache/spark/sql/store/BugTest.scala

```diff
@@ -16,312 +16,321 @@
+        "field8   string," +
+        "field9   string," +
+        "field10   string," +
+        "field11   string," +
+        "field12   string," +
+        "field13   string," +
+        "field14   string," +
+        "field15   string," +
+        "field16   string," +
+        "field17   string," +
+        "localfield8   string," +
+        "localfield9   string," +
+        "localfield10   string," +
+        "localfield11   string," +
+        "localfield15   string," +
+        "localfield16   string," +
+        "field18   string," +
+        "field19   string," +
+        "sfield13   string," +
+        "field20   string," +
+        "field21   string," +
+        "field22   string," +
+        "field23   string )"
+
+    val table2 = "create table tab7 (" +
+        "globalfield16 string," +
+        "globalfield16description string," +
+        "localfield16 string, " +
+        "localfield16description string," +
+        "field20 string)"
 
-    snc1.sql(s"create table test (id  integer," +
-        s" name STRING) using column")
-    snc1.sql("insert into test values (1, 'name1')")
-    snc1.sql(s"GRANT select ON TABLE  test TO  $user2")
+    val table3 = "create table field29_hier (" +
+        "field38 string," +
+        "field39 string," +
+        "field40 string," +
+        "field41 string," +
+        "field42 string," +
+        "field43 string," +
+        "subfield38 string," +
+        "subfield39 string," +
+        "field44 string," +
+        "field45 string," +
+        "field46 string)"
 
-    // TODO : Use the actual connection pool limit
-    val limit = 500
-    for (_ <- 1 to limit) {
-      val snc2 = snc.newSession()
-      snc2.snappySession.conf.set(Attribute.USERNAME_ATTR, user2)
-      snc2.snappySession.conf.set(Attribute.PASSWORD_ATTR, user2)
+    stmt.execute(table1)
+    stmt.execute(table2)
+    stmt.execute(table3)
 
+    val view = "CREATE or replace view view2 as (SELECT " +
+        "A.field9,first(A.field8) as field8,A.field14, A.localfield16," +
+        "A.field20,  A.field11,A.field3, A.field19, " +
+        " first(A.field15) as field15," +
+        " first(A.field23) as field23, first(A.field10) as field10," +
+        "SUM(A.field13 ) as field13,  " +
+        "first(B.globalfield16Description) as globalfield16Description," +
+        " first(C.field44) as field44," +
+        " first(C.field38) as field38,  " +
+        "first(C.field45) as field45," +
+        " first(C.subfield38) as subfield38, " +
+        "first(C.field40) as field40" +
+        " FROM tab1 A  LEFT JOIN tab7 B " +
+        " ON A.field20 = B.field20 AND " +
+        " B.globalfield16 = A.localfield16 LEFT JOIN field29_hier C ON " +
+        " A.field3 = field42 WHERE A.localfield16 ='0L' " +
+        " GROUP BY A.field14, A.field19, " +
+        "A.field9, A.field20, A.field11, A.field3, A.localfield16 " +
+        "having ( SUM(A.field13 ) > 0.001F or SUM(A.field13 ) < -0.001F) );"
 
```

Comment:
The table and view looks the same as in SNAP-2342 test above. Can you move the two into a common method instead of duplicating?